wire/msgmixcm.go

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2023 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package wire
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/decred/dcrd/chaincfg/chainhash"
+)
+
+// MsgMixCM contains a partially-signed mix transaction, with signatures
+// contributed from the peer identity.  When all CM messages are received,
+// signatures can be merged and the transaction may be published, ending a
+// successful mix session.
+//
+// It implements the Message interface.
+type MsgMixCM struct {
+	Signature [64]byte
+	Identity  [33]byte
+	SessionID [32]byte
+	Expiry    int64
```

Comment:
settling on uint32 seems like the best way to deal with this, since tx expiry is already a uint32, and block headers also use uint32 for their heights (even though, internally, dcrd uses int64 quite frequently when dealing with block heights).

Reply:
The `uint32` sounds good and is consistent with the header.

For reference, using `int64` internally was inherited from btcd and really should have all been updated to `uint32` when the header was modified to include the height as a `uint32` too.  In hindsight, there was never really a good reason for it to be an `int64` to begin with and some of the code in there is actually fairly brittle because it is one.